UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -0,0 +1,91 @@
+        private readonly IOptions<FileShareServiceConfiguration> fileShareServiceConfig;
+        private readonly ILogger<FileShareService> logger;
+        private string batchExpiryDateTime;
+
+        public FileShareService(IFileShareServiceClient fileShareServiceClient,
+                                IAuthTokenProvider authTokenProvider,
+                                IOptions<FileShareServiceConfiguration> fileShareServiceConfig,
+                                ILogger<FileShareService> logger)
+        {
+            this.fileShareServiceClient = fileShareServiceClient;
+            this.authTokenProvider = authTokenProvider;
+            this.fileShareServiceConfig = fileShareServiceConfig;
+            this.logger = logger;            
+        }
+        public async Task<CreateBatchResponse> CreateBatch()
+        {
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            var jwtSecurityToken = new JwtSecurityToken(accessToken);
+            var oid = jwtSecurityToken.Claims.FirstOrDefault(m => m.Type == "oid").Value;
+            var uri = $"/batch";
+
+            CreateBatchRequest createBatchRequest = new CreateBatchRequest
+            {
+                BusinessUnit = fileShareServiceConfig.Value.BusinessUnit,
+                Attributes = new List<KeyValuePair<string, string>>()
+                {
+                    new KeyValuePair<string, string>("Exchange Set Type", "Update"),
+                    new KeyValuePair<string, string>("Media Type", "Zip"),
+                    new KeyValuePair<string, string>("Product Type", "AVCS")
+                },
+                ExpiryDate = DateTime.UtcNow.AddDays(1).ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
+                Acl = new Acl()
+                {
+                    ReadUsers = new List<string>() { oid }
+                }
+            };
+            batchExpiryDateTime =createBatchRequest.ExpiryDate;
+
+            string payloadJson = JsonConvert.SerializeObject(createBatchRequest);
+
+            var httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            CreateBatchResponse createBatchResponse = await CreateBatchResponse(httpResponse);
+            return createBatchResponse;                               
+        }
+
+        private async Task<CreateBatchResponse> CreateBatchResponse(HttpResponseMessage httpResponse)
+        {
+            var createBatchResponse = new CreateBatchResponse();
+            var body = await httpResponse.Content.ReadAsStringAsync();
+            if (httpResponse.StatusCode != HttpStatusCode.Created)
+            {
+                logger.LogError(EventIds.FSSCreateBatchNonOkResponse.ToEventId(), $"File share service create batch endpoint responded with {httpResponse.StatusCode} and message {body}");
```

Comment:
Do we plan to have separate event ids for non OK response of each FSS endpoints? For SCS we used a common one.

Reply:
Yes for FSS we are planning to use separate event ids for each FSS endpoint as in future we will be using search batch, create file endpoint in ESS

Reply:
ok